main.tf

```diff
@@ -51,7 +51,7 @@ resource "equinix_network_device" "this" {
   hostname             = var.hostname
   type_code            = data.equinix_network_device_type.this.code
   package_code         = var.software_package
-  version              = data.equinix_network_device_software.this.version
+  version              = var.software_version
```

Comment:
Introducing a variable with no default value would be a breaking change for this module which has a semantic version >= 1.0.0.

The link between `data.equinix_network_device_software.this.version` and `equinix_network_device.this` would also be broken.

`equinix_network_device_software` supports a `version_regex` field.  It appears that this could be used to avoid breaking the API of this module and maintain the integrity between resources.
https://github.com/equinix/terraform-provider-equinix/blob/master/equinix/data_source_network_device_software.go#L113-L158


 ```hcl
data "equinix_network_device_software" "this" {
  device_type    = data.equinix_network_device_type.this.code
  packages       = [var.software_package]
  stable         = true
  version_regexp = var.software_version_regexp
  most_recent    = true
}
```

In the definition of the variable, we should set a default to maintain backwards compatibility.  I think `null` will suffice here (which will avoid the field behavior), but `.*` may be more obvious.

```hcl
variable "software_version_regexp" {
  description = "Device software version regexp (ex: '^9\.0\.4$')"
  type        = string
  default     = null
}
```

Reply:
Thanks for the feedback. Made the changes as you suggested. Please review.